packages/next-app/src/app/pool/[poolId]/(components)/MetadataAttributesTable.tsx

```diff
@@ -111,14 +117,63 @@ function Row({ data }: { data: PoolMetadataAttribute }) {
   );
 }
 
-export function MetadataAttributesTable({ poolId }: { poolId: string }) {
-  const { metadata } = useContext(PoolMetadataContext);
+export function MetadataAttributesTable({ poolId }: { poolId: `0x${string}` }) {
+  const { metadata, handleSetMetadata, handleSetPool } =
+    useContext(PoolMetadataContext);
+
+  handleSetPool(poolId);
+
+  useEffect(() => handleSetMetadata([]), []);
```

Comment:
could you elaborate on what the purpose of this is?

Reply:
the `handleSetPool(poolId);` makes the _selected sidebar pool item_ change the color if we refresh the page. 

and the effect is to avoid the metadata from a pool be shared with a pool that does not have metadata. I'll explain why:

we set the metadata with 

```js
useEffect(() => {
    async function fetchData() {
      const response = await fetch(
        `https://ipfs.io/ipfs/${metadataPool?.metadataCID}`,
        {
          method: "GET",
        }
      );
      const json = await response.json();
      handleSetMetadata(json);
    }
    if (metadataPool?.metadataCID) {
      fetchData();
    }
  }, [metadataPool?.metadataCID]);
```

So if we select a pool with metadata, and then select a pool without metadata, the global state is shared and the pool without metadata gets the metadata from the last pool selected.

To avoid this I created an effect that set the metadata state to an empty array every time that we render the `MetadataAttributesTable` component.

What do you think about that?